src/components/AddItemForm.js

```diff
@@ -37,11 +37,11 @@ const Form = ({token}) => {
                     let documentData = doc.data();
                     fullCollection.push(documentData.itemName);
                 });
-                return fullCollection;
+                setShoppingListCollection(fullCollection);
             }).catch((error) => {
                 console.log("Error getting document:", error);
             });
-    };
+    }, []);
```

Comment:
would love some discussion on it on Sunday if possible, remind me again the purpose

Reply:
The purpose of the empty array is to have a "default" value for  useEffect. This triggers useEffect only after the first render, to prevent overloading requests to the Firebase API.

Reply:
I think we should put in the dependency array: `[token, shoppingListCollection]`. This will run the useEffect if the shoppingListCollection updates or the token updates (i know the token probably won't really update, but my vs code eslint is yelling at me to include it in the array.  : )). Then...on line 49...see my comment.

Reply:
Thanks for that tip, Lauren. I now actually understand what the dependency array is used for. Completely makes sense now.